apps/dashboard/src/main/java/com/akto/utils/RiskScoreOfCollections.java

```diff
@@ -0,0 +1,213 @@
+                Bson filterQSampleData = Filters.and(
+                    Filters.eq("_id.apiCollectionId",((BasicDBObject) basicDBObject.get("_id")).getInt("apiCollectionId")),
+                    Filters.eq("_id.method", ((BasicDBObject) basicDBObject.get("_id")).getString("method")),
+                    Filters.eq("_id.url", ((BasicDBObject) basicDBObject.get("_id")).getString("url"))
+                );
+                bulkUpdatesForApiInfo.add(new UpdateManyModel<>(filterQSampleData, Updates.set(ApiInfo.IS_SENSITIVE, true), new UpdateOptions()));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        if (bulkUpdatesForApiInfo.size() > 0) {
+            ApiInfoDao.instance.getMCollection().bulkWrite(bulkUpdatesForApiInfo, new BulkWriteOptions().ordered(false));
+        }
+
+        AccountSettingsDao.instance.getMCollection().updateOne(
+            AccountSettingsDao.generateFilter(),
+            Updates.set((AccountSettings.RISK_SCORE_TIMERS + "."+ CronTimers.LAST_UPDATED_SENSITIVE_MAP), Context.now()),
+            new UpdateOptions().upsert(true)
+        );
+    }
+
+    private static void updatesForNewEndpoints(int timeStampFilter){
+        List<String> sensitiveInResponse = SingleTypeInfoDao.instance.sensitiveSubTypeInResponseNames();
+        sensitiveInResponse.addAll(SingleTypeInfoDao.instance.sensitiveSubTypeNames());
+        writeUpdatesForSensitiveInfoInApiInfo(sensitiveInResponse, timeStampFilter);
+    }
+
+    public void mapSensitiveSTIsInApiInfo(int timeStampFilter,int cronTime){
+        List<String> sensitiveInResponse = new ArrayList<>();
+        if(timeStampFilter == 0){
+            sensitiveInResponse = SingleTypeInfoDao.instance.sensitiveSubTypeInResponseNames();
+            sensitiveInResponse.addAll(SingleTypeInfoDao.instance.sensitiveSubTypeNames());
+        }else{
+            Bson filter = Filters.gte("timestamp", timeStampFilter);
+            List<AktoDataType> updatedTypes = AktoDataTypeDao.instance.findAll(filter);
+            List<CustomDataType> updaCustomDataTypes = CustomDataTypeDao.instance.findAll(filter);
+            for(AktoDataType dataType: updatedTypes){
+                if (dataType.getSensitiveAlways() || dataType.getSensitivePosition().contains(SingleTypeInfo.Position.RESPONSE_HEADER) || dataType.getSensitivePosition().contains(SingleTypeInfo.Position.RESPONSE_PAYLOAD)) {
+                    sensitiveInResponse.add(dataType.getName());
+                }
+            }
+
+            for(CustomDataType dataType: updaCustomDataTypes){
+                if (dataType.isSensitiveAlways() || dataType.getSensitivePosition().contains(SingleTypeInfo.Position.RESPONSE_HEADER) || dataType.getSensitivePosition().contains(SingleTypeInfo.Position.RESPONSE_PAYLOAD)) {
+                    sensitiveInResponse.add(dataType.getName());
+                }
+            }
+        }
```

Comment:
expose this piece of code as a method in SingleTypeInfoDao

sensitiveInResponse = SingleTypeInfoDao.instance.sensitiveSubTypeInResponseNames();
sensitiveInResponse.addAll(SingleTypeInfoDao.instance.sensitiveSubTypeNames());

if (timeStampFilter > 0) {
    sensitiveInResponse = // call your exposed method in SingleTypeInfoDao
}

Reply:
modified the case as this code was failing cases